### ISSUE TITLE
Prettier to ^1.18.0 conflicts with Typescript 3.2.1

### ISSUE THREAD

**Issue by cartogram** [Shopify/quilt#1123]
**TLDR: Updating Prettier to the latest minor version requires us to be on Typescript 3.3 or later.**

When attempting to update to the latest minor version of Prettier, I encountered an issue where the formatted TypeScript output added a trailing comma to `Tuples`. 

For example

```ts
// Before
export function useSomeHook(): [
  string,
  string
] { /*...*/}

// After 
export function useSomeHook(): [
  string,
  string, <-- Trailing comma 
] { /*...*/}
```

This is a problem because our version of Typescript does not support trailing commas on Tuples. It looks like they were were added in Typescript 3.3 via this [PR](https://github.com/Microsoft/TypeScript/issues/28893).

Work has already been started in [this PR to upgrade Typescript](https://github.com/Shopify/quilt/pull/962/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R71). When that is shipped, we can upgrade to the latest prettier.
